src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -865,6 +865,25 @@ winrt::Microsoft::Terminal::Settings::Model::ColorScheme CascadiaSettings::GetCo
     return _globals->ColorSchemes().TryLookup(schemeName);
 }
 
+// Method Description:
+// - updates all references to that color scheme with the new name
+// Arguments:
+// - oldName: the original name for the color scheme
+// - newName: the new name for the color scheme
+// Return Value:
+// - <none>
+void CascadiaSettings::UpdateColorSchemeReferences(const hstring oldName, const hstring newName)
+{
+    // update all profiles referencing this color scheme
+    for (auto profile : _allProfiles)

```

Comment:
`const auto&` :smile:

Reply:
Fixed. So, here's what's confusing to me about this though. `const auto&` means we're taking an "immutable reference to the variable", but a few lines down, we're modifying that variable. So then is `const auto&` have the sole purpose of not incrementing the ref-count?

Reply:
Oh. Right. It's _complicated_. WinRT objects _do not have constness_. It isn't part of WinRT.

Therefore, we're just using the C++ rules of constness: the projection object isn't gonna change, but the thing inside it absolutely is.

Reply:
If it builds this way. Maybe it wont'?